src/database/postgres.go

```diff
@@ -0,0 +1,198 @@
+package database
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	pb "accretional.com/semantifly/proto/accretional.com/semantifly/proto"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgconn"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type PgxIface interface {
+	Begin(context.Context) (pgx.Tx, error)
+	Close(context.Context) error
+	Query(ctx context.Context, sql string, args ...interface{}) (pgx.Rows, error)
+	Exec(ctx context.Context, sql string, arguments ...interface{}) (pgconn.CommandTag, error)
+}
+
+func establishConnection(ctx context.Context) (*pgx.Conn, error) {
+	conn, err := pgx.Connect(ctx, os.Getenv("DATABASE_URL"))
+	if err != nil {
+		return nil, fmt.Errorf("unable to connect to database: %w", err)
+	}
+	return conn, nil
+}
+
+func initializeDatabaseSchema(ctx context.Context, conn PgxIface) error {
+    tx, err := conn.Begin(ctx)
+    if err != nil {
+        return fmt.Errorf("unable to connect to database: %w", err)
+    }
+    defer tx.Rollback(ctx)
+
+    // Create the main table
+    _, err = tx.Exec(ctx, `
+        CREATE TABLE IF NOT EXISTS index_list (
+            name TEXT PRIMARY KEY,
+            uri TEXT,
+            data_type TEXT,
+            source_type TEXT,
+            first_added_time TIMESTAMP,
+            last_refreshed_time TIMESTAMP,
+            content TEXT,
+            search_vector tsvector,
+            word_occurrences JSONB
+        )
+    `)
+    if err != nil {
+        return fmt.Errorf("failed to create table: %w", err)
+    }
+
+    // Create indexes
+    _, err = tx.Exec(ctx, `
+        CREATE INDEX IF NOT EXISTS idx_search_vector ON index_list USING GIN (search_vector);
+        CREATE INDEX IF NOT EXISTS idx_word_occurrences ON index_list USING GIN (word_occurrences);
+    `)
+    if err != nil {
+        return fmt.Errorf("failed to create indexes: %w", err)
+    }
+
+    // Create materialized view
+    _, err = tx.Exec(ctx, `
+        CREATE MATERIALIZED VIEW IF NOT EXISTS mv_index_list AS
+        SELECT * FROM index_list
```

Comment:
What is the point of this materialized view if it's almost exactly the same as index_list?

Reply:
Materialised view maintains a copy of the query result which will be used for faster query (instead of computing it every time)

Reply:
Removed it